### PR TITLE
Stop flushing the stored messages stream when the connection dies,

### PR DESCRIPTION
**Pull request by miotabbott**
the stream can get flushed multiple times in parrallel otherwise.

### CODE REVIEW

File: test/client.js [hunk 1]
```diff
@@ -252,5 +252,70 @@ describe('MqttClient', function () {
...
+            return true
+          } else {
+            return false
+          }
+        })
+        if (duplicateId) {
+          done(new Error('Multiple duplicate acked subscriptions received for messageId ' + duplicateId))
+        } else {
+          done()
+        }
```
Comment (mcollina) [inline test/client.js]: we should close the server before calling done.

File: test/client.js [hunk 2]
```diff
@@ -252,5 +252,70 @@ describe('MqttClient', function () {
...
+        client.end(true)
+      }, 3000)
```
Comment (mcollina) [inline test/client.js]: I am really scared about timeouts in tests, as they are often a source of flakyness.
Can we remove it?

File: test/client.js [hunk 3]
```diff
@@ -252,5 +252,70 @@ describe('MqttClient', function () {
...
+      setTimeout(function () {
+        var duplicateId
+        Object.keys(subIds).some(function (messageId) {
+          if (subIds[messageId] > 1) {
+            duplicateId = subIds[messageId]
+            return true
+          } else {
+            return false
+          }
+        })
```
Comment (mcollina) [inline test/client.js]: Why are you using `some`, if you are not storing the result in  `duplicateId`?

**Comment by miotabbott** [comment]
@mcollina requested changes made.

**Comment by mcollina** [comment]
Thanks, releasing soon. Feel free to update your PR for v1.

**Comment by miotabbott** [comment]
Thanks! Unfortunately a mistake slipped in with the last change. #469 is up to fix it.

**Pull request merged**